processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -0,0 +1,200 @@
+      @JsonProperty("lowerStrict") Boolean lowerStrict,
+      @JsonProperty("upperStrict") Boolean upperStrict,
+      @JsonProperty("alphaNumeric") Boolean alphaNumeric
+  )
+  {
+    this.dimension = Preconditions.checkNotNull(dimension, "dimension can not be null");
+    Preconditions.checkState((lower != null) || (upper != null), "lower and upper can not be null at the same time");
+    this.upper = upper;
+    this.lower = lower;
+    this.lowerStrict = (lowerStrict == null) ? false : lowerStrict;
+    this.upperStrict = (upperStrict == null) ? false : upperStrict;
+    this.alphaNumeric = (alphaNumeric == null) ? false : alphaNumeric;
+  }
+
+  @JsonProperty
+  public String getDimension()
+  {
+    return dimension;
+  }
+
+  @JsonProperty
+  public String getUpper()
+  {
+    return upper;
+  }
+
+  @JsonProperty
+  public String getLower()
+  {
+    return lower;
+  }
+
+  @JsonProperty
+  public boolean isLowerStrict()
+  {
+    return lowerStrict;
+  }
+
+  @JsonProperty
+  public boolean isUpperStrict()
+  {
+    return upperStrict;
+  }
+
+  @JsonProperty
+  public boolean isAlphaNumeric()
+  {
+    return alphaNumeric;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] dimensionBytes = StringUtils.toUtf8(this.getDimension());
+    byte[] lowerBytes = this.getLower() == null ? new byte[0] : StringUtils.toUtf8(this.getLower());
+    byte[] upperBytes = this.getUpper() == null ? new byte[0] : StringUtils.toUtf8(this.getUpper());
+    byte lowerStrictByte = (this.isLowerStrict() == false) ? (byte) 0 : (byte) 1;
+    byte upperStrictByte = (this.isUpperStrict() == false) ? (byte) 0 : (byte) 1;
+    byte AlphaNumericByte = (this.isAlphaNumeric() == false) ? (byte) 0 : (byte) 1;
+
+    if (this.getLower() == null) {
+      ByteBuffer boundCacheBuffer = ByteBuffer.allocate(
+          6
+          + dimensionBytes.length
+          + upperBytes.length
+      );
+      boundCacheBuffer.put(DimFilterCacheHelper.BOUND_CACHE_ID)
+                        .put(dimensionBytes)
+                        .put((byte) 0xFF)
+                        .put(upperBytes)
+                        .put((byte) 0xFF)
+                        .put(upperStrictByte)
+                        .put((byte) 0xFF)
+                        .put(AlphaNumericByte);
+      return boundCacheBuffer.array();
+
+    } else if (this.getUpper() == null) {
+      ByteBuffer betweenCacheBuffer = ByteBuffer.allocate(
+          6
+          + dimensionBytes.length
+          + upperBytes.length
+      );
+      betweenCacheBuffer.put(DimFilterCacheHelper.BOUND_CACHE_ID)
+                        .put(dimensionBytes)
+                        .put((byte) 0xFF)
+                        .put(lowerBytes)
+                        .put((byte) 0xFF)
+                        .put(lowerStrictByte)
+                        .put((byte) 0xFF)
+                        .put(AlphaNumericByte);
+      return betweenCacheBuffer.array();
+
+    } else {
```

Comment:
after taking care of nulls to EMPTY bytes array, why do we need to handle the null cases for getUpper/Lower ?
what goes wrong we do not put 0xFF as separators ? (puting them for no use just increases the size of resulting byte[])